### PR TITLE
Add c:JSONAPI.View.get_field/3

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes https://github.com/beam-community/jsonapi/issues/272.

### CODE REVIEW

File: test/jsonapi/view_test.exs
Comment (mattpolzin) [inline test/jsonapi/view_test.exs:314]: Let's make this test differentiate values coming from `get_field/3` vs. `static_fun/2` so we are testing that `static_fun/2` is indeed called. The same goes for the `static_field` property. If the stored value of `static_field` and the generated value of `static_fun/2` are changed slightly then it will go to prove that `get_field/3` is being called for every field _except_ for `:static_fun` which is what I expect based on the code.

File: test/jsonapi/view_test.exs
Comment (mattpolzin) [inline test/jsonapi/view_test.exs:53]: This is supposed to be the 2-arity function that gets called when defined with `get_field/3` as a fallback right?

**Review by mattpolzin — COMMENTED** [review]
Just have some ideas around the test.

File: test/jsonapi/view_test.exs
Comment (whatyouhide) [reply test/jsonapi/view_test.exs:53]: Ah, yes 🤦 Fixed.

File: test/jsonapi/view_test.exs
Comment (whatyouhide) [reply test/jsonapi/view_test.exs:314]: Yep, good call. Updated.

**Review by mattpolzin — APPROVED** [review]
Nice!

My approval isn't official since I am not a maintainer, but this looks good to me once tests pass (only maintainers can trigger the CI tests to run).

**Comment by whatyouhide** [comment]
Ping @davydog187 and/or @doomspork? 🙃

**Comment by mattpolzin** [comment]
@doomspork Would you be interested in having me help maintain (review and/or release) this project? I'm very interesting in chipping in more formally and keeping momentum up. Sorry, I feel a comment on a PR is a terrible place to make that plea, I've just had trouble finding a better way to reach out!

**Comment by mattpolzin** [comment]
@whatyouhide do you mind merging the main branch into yours again to see if GitHub will recognize that it hasn't run CI against this branch yet? I now have the ability to merge this PR once we figure out why GitHub has forgotten that there are CI actions to run against it!

**Comment by doomspork** [comment]
@mattpolzin you'll need to re-approve as well 👍

**Review by mattpolzin — APPROVED** [review]
Re-approving (pending us getting CI to run).

**Comment by whatyouhide** [comment]
@mattpolzin just rebased on `master` and pushed.

**Comment by mattpolzin** [comment]
Thank you! I am fixing CI over [here](https://github.com/beam-community/jsonapi/pull/279) so I'll have you merge once more after that lands. Appreciate your patience!

**Comment by mattpolzin** [comment]
Ok, that's sorted when you get the chance to merge master in again @whatyouhide .

**Comment by davydog187** [comment]
Sorry all, just catching up here. Let me know if I can help @whatyouhide @mattpolzin

**Comment by doomspork** [comment]
Updated the branch protections which should now allow us to update a branch to master 👍 Just did so here and it has kicked off CI.

**Review by doomspork — APPROVED** [review]

**Pull request merged**
